x-pack/plugins/rule_registry/scripts/generate_ecs_fieldmap/index.js

```diff
@@ -32,19 +32,27 @@ async function generate() {
 
   const flatYaml = await yaml.safeLoad(await readFile(ecsYamlFilename));
 
-  const fields = mapValues(flatYaml, (description) => {
-    const field = {
-      type: description.type,
-      array: description.normalize.includes('array'),
-      required: !!description.required,
-    };
+  const fields = reduce(
+    flatYaml,
+    (fieldsObj, value, key) => {
+      const field = {
+        type: value.type,
+        array: value.normalize.includes('array'),
+        required: !!value.required,
+      };
 
-    if (description.scaling_factor) {
-      field.scaling_factor = description.scaling_factor;
-    }
+      if (value.scaling_factor) {
+        field.scaling_factor = value.scaling_factor;
+      }
 
-    return field;
-  });
+      if (field.type !== 'constant_keyword') {
```

Comment:
Should we be throwing an error if we see constant_keyword fields instead of just not adding them?

Reply:
We could potentially log an error, but throwing here would have the effect of breaking the script (since ECS _does_ have these field definitions). Expanding on your point, though: we aren't going to want to run this script again in the future without further work (either disallowing non-additive changes, or providing a migration mechanism).